New Doc Style Guide (#85)
pyansys/dev-guide: doc/source/doc-style/docstrings.rst

These sections should follow numpydoc style. To avoid inconsistencies between
PyAnsys libraries, adhere to the additional style guidelines that follow.

Short Summary
-------------

> This is a single-line that goes right after the declaration of the class or
> function for briefly describing what the class or function or does. The
> `short summary` is mandatory. If it is not present, :ref:`Doc Style Tools` will
> raise an error.

Reviewer: ```suggestion
This is a single line that goes immediately after the declaration of the class
or function to briefly describe what the class or function does. The
`short summary` is mandatory. If it is not present, :ref:`Doc Style Tools` will
raise an error.
```

> This is a single-line which goes right after the declaration of the function or
> class and provides a quick overview about the final goal of the code. The
> ``short summary`` is mandatory. If not present, :ref:`Doc Style Tools` will
> raise an error.

Reviewer: ```suggestion
This is a single-line that goes right after the declaration of the function or
class to provide a brief description of what the function or code does. The
`short summary`` is mandatory. If it is not present, :ref:`Doc Style Tools` will
raise an error.
```

The short summary can be declared on the same line as the opening quotes or on
the next line. While `PEP 257
<https://peps.python.org/pep-0257>`_ accepts both ways, you must be consistent across your
project. If you decide to declare the short summary on the same line,

> refer to :ref:`Numpydoc Validation` because the ``"GL01"`` check needs to be

Reviewer: ```suggestion
refer to :ref:`Numpydoc Validation` because the ``"GL01"`` check must be
```

> The short summary can be declared on the same line as the opening quotes or on
> the next line. Both ways are accepted by `PEP 257
> <https://peps.python.org/pep-0257>`_ but you must be consistent across your
> project. In case you decide to declare the short summary on the same line,
> please refer to :ref:`Numpydoc Validation`, as ``"GL01"`` check needs to be
> disabled.

Reviewer: ```suggestion
The short summary can be declared on the same line as the opening quotes or on
the next line. While `PEP 257
<https://peps.python.org/pep-0257>`_ accepts both ways, you must be consistent across your
project. If you decide to declare the short summary on the same line,
refer to :ref:`Numpydoc Validation` because the ``"GL01"`` check needs to be
disabled.
```

> Depending in whether you are documenting a ``Class`` or a ``function``, you will

Reviewer: ```suggestion
Depending 0n whether you are documenting a ``Class`` or a ``function``, you will
```

just the GET method. Examples should be included to demonstrate usage.

Parameters
----------

> Both classes and methods have parameters in their function signatures. All parameters in a function

Author: The only difference between a function and a class method is that the second one is a function linked to a class.

I agree the previous line is confusing. Instead, it could read like this:

```suggestion
Functions and class methods may have parameters in their signatures. All these
parameters should be documented in the ``Parameters`` section.
```

  
Returns
-------
The ``Returns`` section contains only the return data type and a brief description
that concludes with a period:

.. code-block:: rst

  Returns
  -------
  dict
      Dictionary of components with their absolute paths.
 

A class does not have a ``Returns`` section. If a ``Boolean`` is returned, format the
``Returns`` section like this:

.. code-block:: rst

  Returns
  -------
  bool
      ``True`` when successful, ``False`` when failed.

It is possible for the ``Returns`` section to look like the ``Parameters`` one
if variable names are provided:

.. code-block:: rst

  Returns
  -------
  has_succeeded : bool
      ``True`` when successful, ``False`` when failed.

It is possible for more than one item to be returned:

.. code-block:: rst

  Returns
  -------
  type
      Ground object.
  str
      Ground name.

If a method does not have a decorator, the basic implementation of Python
methods is used. In this case, while ``None`` is returned, you do not document it.
Consequently, such a method does not have a 'Returns' section.

Examples
--------

The ``Examples`` section provides a quick reference on how to use a method or

> function. This section needs to be compliant with the `doctest

Reviewer 2: ```suggestion
a function. This section needs to be compliant with the `doctest
```

consider the following function:

.. code-block:: rst

   Examples
   --------
   Create an instance of HFSS and connect to an existing HFSS
   design or create a new HFSS design if one does not exist.

   >>> from pyaedt import Hfss
   >>> hfss = Hfss()
   pyaedt info: No project is defined...
   pyaedt info: Active design is set to...

Notice that if the definition of the function gets updated, this
section needs to be updated too.

Additional Directives
=====================
Since Python docstrings are written using RST syntax, it is possible to take
advantage of some directives available in this Markup language. Among those, it
is possible to find:

- ``.. note::`` directive is useful for highlighting important
  information once the documentation gets rendered.

- ``.. warning::`` is usually used to point out an action that might result in
  data loss.

- ``.. deprecated:: X.Y.Z`` to inform the user about the deprecated status of
  the object or functionality.

You can find additional information and examples at `numpydoc`_. Reference
this documentation as the primary source regarding docstring styles for directives
that are not covered here.

Example
=======

> An generic docstring example compliant with PyAnsys guidelines is shown below:

Reviewer 2: ```suggestion
A generic docstring example compliant with PyAnsys guidelines is shown below:
```
